### PR TITLE
fix(tools): account for custom componentsSubpath in router

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Account for the `options.componentSubpath` in dev-server router.
2. Refactored dev server router to clarify intent.

TODO:
 
- [x] Add comments to document the redirects inline
- [x] Add changeset

Reference: https://github.com/RedHat-UX/red-hat-design-system/issues/885 

## Testing Instructions

1. Manifest demos should have
    2. urls pointing to the demos (e.g. https://patternflyelements.org/components/jazz-hands)
    3. src urls pointing  to the demo sources (e.g. https://github.com/patternfly/patternfly-elements/elements/pf-jazz-hands/demos/pf-jazz-hands.html)
2. Dev server URLs should correspond to the docs site URLs, and not to the file structure

## Notes to Reviewers

1.

### CODE REVIEW

**Comment by brianferry** [comment]
Built this branch & copied the `tools/pfe-tools/dev-server/config.js` into my rhds main branch & seeing the dev server working now

**Comment by zeroedin** [comment]
The changes so far break the docs.  When `componentSubpath` is set to `elements` either as the default or via `.pfe.config.json` The value of `componentSubpath`  is used to copy demo files into `_site/elements/element/demo/index.html`.  The files end up in `elements` dir instead of `components` dir as specified in the docs url structure.  Still working my way through this issue and potential fixes however,  I'm starting to wonder if it makes sense to separate out config for the devServer and docs sites.
```
componentSubpath:  "elements",
docsComponentSubpath:  "components"
```

File: tools/pfe-tools/config.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ interface SiteOptions {
   stylesheets?: string[];
   /** Title for main page of the demo */
   title?: string;
-  /** Site subpath for components. e.g. 'elements'. default: 'components' */
+  /** Site subpath for components. e.g. 'elements'. default: 'elements' */
```
Comment (zeroedin) [inline tools/pfe-tools/config.ts:16]: Revert comment back to default of components
```Suggestion
/** Site subpath for components. e.g. 'elements'. default: 'components' */
```

File: tools/pfe-tools/dev-server/config.ts [hunk 2]
```diff
@@ -153,6 +149,7 @@ function pfeDevServerPlugin(options: PfeDevServerInternalConfig): Plugin {
   };
 }
 
+
```
Comment (zeroedin) [inline tools/pfe-tools/dev-server/config.ts:152]: Remove whitespace
```Suggestion
```

**Comment by heyMP** [comment]
@brianferry I'm getting build errors when trying to link `@patternfly/pfe-tools`.  I'm probably not linking it correctly.  Can you try again?

```bash
➜  mpotter-mac as michael  red-hat-design-system (main) ✗
   npm run build

> @rhds/elements@1.0.0 build
> wireit

🏃 [bundle] Running command "node scripts/bundle.js"
🏃 [analyze] Running command "cem analyze"
🏃 [compile] Running command "ttsc"
file:///Users/michael/Projects/red-hat-design-system/node_modules/@patternfly/pfe-tools/custom-elements-manifest/lib/Manifest.js:5
import { deslugify } from '@patternfly/pfe-tools/config.js';
         ^^^^^^^^^
SyntaxError: The requested module '@patternfly/pfe-tools/config.js' does not provide an export named 'deslugify'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:123:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:189:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
    at async importConfig (/Users/michael/Projects/red-hat-design-system/node_modules/@web/config-loader/src/importConfig.js:28:20)
    at async getUserConfig (file:///Users/michael/Projects/red-hat-design-system/node_modules/@custom-elements-manifest/analyzer/src/utils/cli.js:41:18)
    at async file:///Users/michael/Projects/red-hat-design-system/node_modules/@custom-elements-manifest/analyzer/index.js:34:24
^C💀 [analyze] Killed
💀 [compile] Killed
💀 [bundle] Killed
```

File: tools/pfe-tools/config.ts [hunk 3]
```diff
@@ -46,6 +46,7 @@ const SITE_DEFAULTS: Required<SiteOptions> = {
 const DEFAULT_CONFIG: PfeConfig = {
   demoURLPrefix: 'https://patternflyelements.org/',
   sourceControlURLPrefix: 'https://github.com/patternfly/patternfly-elements/tree/main/',
+  elementsDir: 'elements',
```
Comment (heyMP) [inline tools/pfe-tools/config.ts:49]: I could see us needing this to be an array in the future.  I.e.  ["elements", "patterns", "apps"].  I think we could add this as a non-breaking feature in another release though.

**Comment by zeroedin** [comment]
Notes from Office hours discussion on this topic.

Intent is to have dev-server and docs site URLs to line up so port number or full your could easily be swapped.  The configuration isn't specific to project folder names only to the end points in the URL structure and is set by a docs specific nature.  The dev server should adapt to that configuration, not be independent of it.

The changes on this PR will need adjusted to match to this intent.

Example
localhost:8000/components/**
localhost:8080/components/**
patternflyelements.com/components/**

**Comment by zeroedin** [comment]
@bennypowers let me know if the latest changes align now with the intent.  From my understanding now I believe they do.

File: tools/pfe-tools/config.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ interface SiteOptions {
   stylesheets?: string[];
   /** Title for main page of the demo */
   title?: string;
-  /** Site subpath for components. e.g. 'elements'. default: 'components' */
+  /** Site subpath for components. e.g. 'components'. default: 'components' */
```
Comment (bennypowers) [inline tools/pfe-tools/config.ts:16]:
```suggestion
  /** Site subpath for components. default: 'components' i.e. 'https://patternflyelements.org/components' */
```

File: tools/pfe-tools/config.ts (hunk 3 above)
Comment (bennypowers) [reply tools/pfe-tools/config.ts:49]: yeah probably. That would likely use some different kind of manifest though, a "pattern manifest", if you will.

File: tools/pfe-tools/dev-server/config.ts [hunk 5]
```diff
@@ -94,57 +94,52 @@ function pfeDevServerPlugin(options: PfeDevServerInternalConfig): Plugin {
...
+            const { element, fileName } = ctx.params;
+            ctx.redirect(`/${elementsDir}/${element}/${fileName}.ts`);
+          })
+          // Redirect `elements/jazz-hands/*.js` to `elements/pf-jazz-hands/*.ts`
+          .get(`/${elementsDir}/:element/:fileName.js`, async ctx => {
+            const { element, fileName } = ctx.params;
+            ctx.redirect(`/${elementsDir}/${element}/${fileName}.ts`);
+          })
+          // Redirect `components/jazz-hands/demo/*.js|css` to `components/pf-jazz-hands/demo/*.js|css`
+          // If request is `components/jazz-hands/demo/some-other-demo/*.js|css redirect files to `components/pf-jazz-hands/demo/*.js|css`
```
Comment (bennypowers) [inline tools/pfe-tools/dev-server/config.ts:120]:
```suggestion
```

**Comment by bennypowers** [comment]
Needs a changeset

**Comment by eyevana** [comment]
@zeroedin copied the config to my cpelements repo & the demos are working for me again 🥳

File: .changeset/long-islands-retire.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"@patternfly/pfe-tools": patch
+---
+
+`pfe-tools`: updates dev-server router to use project subpath configuration
```
Comment (bennypowers) [inline .changeset/long-islands-retire.md:5]:
```suggestion
**Dev Server**: updates router to use project subpath configuration
```

**Pull request merged**